torch/_inductor/codegen/cpp_micro_gemm.py

```diff
@@ -183,6 +187,36 @@ def get_b_layout(self) -> LayoutType:
         return LayoutType.NORMAL
 
 
+class CppMicroGemmF16Abstract(CppMicroGemm):
+    ALLOCATE_BUFFER = r"""
+  int64_t {{buffer_name}}_dtype_itemsize = (
+    std::is_same_v<{{buffer_dtype}}, at::BFloat16>
+    || std::is_same_v<{{buffer_dtype}}, at::Half>
+    ) ? 2 : 4;
+  auto& {{buffer_name}}_allocator = *at::getCPUAllocator();
+  auto {{buffer_name}}_work_data = {{buffer_name}}_allocator.allocate({{buffer_size}}*{{buffer_name}}_dtype_itemsize);
```

Comment:
Heap allocation for each worker thread may be slower than stack allocation (locking overhead due to multi-threading & manipulation of malloc-internal data structures. Even tcmalloc isn't completely lock-free) and since such a buffer would be allocated multiple times in each worker thread (which may contend with all other worker threads for memory allocator locks), can we use a stack allocated buffer per thread instead, if doing so results in a discernible performance difference for large input shapes? Thanks!

The flex attention codegened kernel uses a heap allocator but it's invoked by only one thread, and then individual worker threads use equal chunks of the allocated buffer(s), so there's no performance penalty for using heap memory there.